cwms-data-api/src/test/java/cwms/cda/api/location/kind/GateChangeStressTest.java

```diff
@@ -0,0 +1,250 @@
+package cwms.cda.api.location.kind;
+
+import com.google.common.flogger.FluentLogger;
+import cwms.cda.api.Controllers;
+import cwms.cda.api.enums.Unit;
+import cwms.cda.api.enums.UnitSystem;
+import cwms.cda.api.errors.NotFoundException;
+import cwms.cda.data.dao.LocationsDaoImpl;
+import cwms.cda.data.dao.location.kind.BaseOutletDaoIT;
+import cwms.cda.data.dao.location.kind.OutletDao;
+import cwms.cda.data.dao.location.kind.ProjectStructureIT;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.Location;
+import cwms.cda.data.dto.LookupType;
+import cwms.cda.data.dto.location.kind.GateChange;
+import cwms.cda.data.dto.location.kind.GateSetting;
+import cwms.cda.data.dto.location.kind.Outlet;
+import cwms.cda.formatters.Formats;
+import fixtures.CwmsDataApiSetupCallback;
+import io.restassured.filter.log.LogDetail;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.Period;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAmount;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.servlet.http.HttpServletResponse;
+import mil.army.usace.hec.metadata.constants.NumericalConstants;
+import mil.army.usace.hec.test.database.CwmsDatabaseContainer;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import usace.cwms.db.jooq.codegen.packages.CWMS_PROJECT_PACKAGE;
+import static cwms.cda.api.Controllers.*;
+import static cwms.cda.data.dao.DaoTest.getDslContext;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.*;
+
+class GateChangeStressTest extends BaseOutletDaoIT {
+
+    private static final FluentLogger LOGGER = FluentLogger.forEnclosingClass();
+    private static final CwmsId PROJECT_ID = new CwmsId.Builder().withOfficeId(OFFICE_ID).withName("PROJECT3").build();
+    private static final Location PROJECT_LOCATION = buildProjectLocation(PROJECT_ID.getName());
+    private static final String CONDUIT_GATE_RATING_SPEC_ID = PROJECT_ID.getName() + ".Opening-ConduitGate,Elev;Flow-ConduitGate.Standard.Production";
+    private static final CwmsId CONDUIT_GATE_RATING_GROUP = new CwmsId.Builder()
+            .withName("Rating-" + PROJECT_ID.getName() + "-ConduitGate")
+            .withOfficeId(OFFICE_ID)
+            .build();
+
+    private static final TemporalAmount INTERVAL = Duration.of(5, ChronoUnit.MINUTES);
+    private static final ZonedDateTime START = ZonedDateTime.of(2020, 1, 1, 0, 0, 0, 0, NumericalConstants.UTC_ZONEID);
+    private static final ZonedDateTime END = START.plusYears(1).plus(INTERVAL);
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        //1 year of 5 minute gate change data for 18 gates
```

Comment:
Who changes a gate every 5 minutes? Or is this just an extreme example to cause problems?

Reply:
This was the most extreme example of something like SCADA data, or some other automated process.

Reply:
Crazy. Thanks for the info.